backend/eventloop/shared_resource_loop/sharedresourceloop.go

```diff
@@ -108,8 +108,8 @@ func (srEventLoop *SharedResourceEventLoop) GetGitopsEngineInstanceById(ctx cont
 // Ensure the user's workspace is configured, ensure a GitOpsEngineInstance exists that will target it, and ensure
 // a cluster access exists the give the user permission to target them from the engine.
 func (srEventLoop *SharedResourceEventLoop) GetOrCreateSharedResources(ctx context.Context,
-	workspaceClient client.Client, workspaceNamespace corev1.Namespace) (*db.ClusterUser,
-	*db.ManagedEnvironment, *db.GitopsEngineInstance, *db.ClusterAccess, error) {
+	workspaceClient client.Client, workspaceNamespace corev1.Namespace) (*db.ClusterUser, bool,
```

Comment:
As above, add a quick comment mentioning that the bools refer to whether the particular object has been created.